client/src/main/java/io/split/client/HttpSplitChangeFetcher.java

```diff
@@ -59,27 +69,68 @@ long makeRandomTill() {
-                throw new IllegalStateException(
-                        String.format("Could not retrieve splitChanges since %s; http return code %s", since, response.statusCode())
-                );
+                    if (response.statusCode() == HttpStatus.SC_BAD_REQUEST && SPEC_VERSION.equals(Spec.SPEC_1_3) && _rootURIOverriden) {
+                        SPEC_VERSION = Spec.SPEC_1_2;
+                        _log.warn("Detected proxy without support for Feature flags spec {} version, will switch to spec version {}",
+                                SPEC_1_3, SPEC_1_2);
+                        _lastProxyCheckTimestamp = System.currentTimeMillis();
+                        continue;
+                    }
+
+                    _telemetryRuntimeProducer.recordSyncError(ResourceEnum.SPLIT_SYNC, response.statusCode());
+                    throw new IllegalStateException(
+                            String.format("Could not retrieve splitChanges since %s; http return code %s", since, response.statusCode())
+                    );
+                }
+                break;
+            } catch (Exception e) {
+                throw new IllegalStateException(String.format("Problem fetching splitChanges since %s: %s", since, e), e);
+            } finally {
+                _telemetryRuntimeProducer.recordSyncLatency(HTTPLatenciesEnum.SPLITS, System.currentTimeMillis() - start);
             }
-            return Json.fromJson(response.body(), SplitChange.class);
-        } catch (Exception e) {
-            throw new IllegalStateException(String.format("Problem fetching splitChanges since %s: %s", since, e), e);
-        } finally {
-            _telemetryRuntimeProducer.recordSyncLatency(HTTPLatenciesEnum.SPLITS, System.currentTimeMillis() - start);
         }
+
+        String body = response.body();
+        if (SPEC_VERSION.equals(Spec.SPEC_1_2)) {
+            body = convertBodyToOldSpec(body);
+            _lastProxyCheckTimestamp = System.currentTimeMillis();
+        }
+        return Json.fromJson(body, SplitChange.class);
+    }
+
+    public Long getLastProxyCheckTimestamp() {
+        return _lastProxyCheckTimestamp;
+    }
+
+    public void setLastProxyCheckTimestamp(long lastProxyCheckTimestamp) {
+        synchronized (_lock) {
+            _lastProxyCheckTimestamp = lastProxyCheckTimestamp;
+        }
+    }
+
+    private String convertBodyToOldSpec(String body) {
+        JsonObject targetBody = Json.fromJson("{\"ff\": {\"t\":-1, \"s\": -1}," +
+                "\"rbs\": {\"d\":[], \"t\":-1, \"s\": -1}}", JsonObject.class);
+        JsonObject jsonBody = Json.fromJson(body, JsonObject.class);
+        targetBody.getAsJsonObject("ff").add("d", jsonBody.getAsJsonArray("splits"));
+        targetBody.getAsJsonObject("ff").add("s", jsonBody.get("since"));
+        targetBody.getAsJsonObject("ff").add("t", jsonBody.get("till"));
+        return Json.toJson(targetBody);
```

Comment:
We can improve this by creating a `SplitChangesOldPayloadDto` class and returning the new DTO here
```suggestion
    private SplitChange returnBodyWithOldSpec(String body) {
        SplitChangesOldPayloadDto oldDto = Json.fromJson(body, SplitChangesOldPayloadDto.class);
        
        return oldDto.ToSplitChanges();
```

Reply:
will do